pkg/pluginregistry/registry.go

```diff
@@ -139,14 +150,15 @@ func (r *PluginRegistry) discoverPlugin(endpoint string) {
 	}
 	plugin.Status = loaded
 	plugin.Info = *resp.ModelInfo
+	plugin.ID = strings.ToLower(fmt.Sprintf("%s-%s", resp.ModelInfo.Name, resp.ModelInfo.Version))
 
```

Comment:
I think that once we find the plugin, we should delete the entry under plugin.Name and re-add under plugin.ID. The name is nothing but a provisional ID.

Reply:
I was going down a similar path yesterday,
thinking that we can ditch the `Port` field in favor of an `Endpoint` field (might get useful if we decide to move away from Sidecars, as no changes would be required), and use this `Endpoint` field as `ID` before we successfully load the plugin info, then update id with the received information.